x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -52,7 +52,7 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
     )
   );
 
-  const inventoryItems = Object.keys(first(results) as any);
+  const inventoryItems = Object.keys(first(results)!);
```

Comment:
Is this a safe assertion? How do we know `results` is not empty?

Would

```suggestion
  const inventoryItems = Object.keys(first(results) ?? {});
```

be safer?

Reply:
`results` always has one result for everything in `criteria`. I can add this to the top of the function:

```js
if (criteria.length === 0) throw new Error('Cannot execute an alert with 0 conditions');
```